drivers/adc/ad7779/ad7779.c

```diff
@@ -425,8 +425,8 @@ int32_t ad7779_set_state(ad7779_dev *dev,
 
 	ret = ad7779_spi_int_reg_write_mask(dev,
 					    AD7779_REG_CH_DISABLE,
-					    AD7779_CH_DISABLE(0x1),
-					    AD7779_CH_DISABLE(state));
+					    AD7779_CH_DISABLE(ch),
+					    state ? AD7779_CH_DISABLE(ch) : 0x00);
```

Comment:
I wouldn't use the hexadecimal representation - for consistency with the rest of code:
state ? AD7779_CH_DISABLE(ch) : 0);